File: boat-spring-boot-sample/src/test/java/sample/java/xyz/srclab/spring/boot/lang/LangSample.java
package sample.java.xyz.srclab.spring.boot.lang;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.test.context.testng.AbstractTestNGSpringContextTests;
import org.testng.Assert;
import org.testng.annotations.Test;
import xyz.srclab.common.codec.aes.AesKeys;
import xyz.srclab.spring.boot.lang.KeyString;

import javax.crypto.SecretKey;

@SpringBootTest(classes = Starter.class)
//@ContextConfiguration(classes = {TestStarter.class})
public class LangSample extends AbstractTestNGSpringContextTests {

    private static final Logger logger = LoggerFactory.getLogger(LangSample.class);

    @Value("AES,BASE64:rliqBhMdiKQDcH8lqNZdIg==")
    private KeyString keyString;

    @Test
    public void testEncodeString() {
        logger.info("encodeString: {}", keyString);
        SecretKey key = AesKeys.newKey("123");
        Assert.assertEquals(keyString.decodeString(key), "some password");
    }
}

File: boat-spring-boot-sample/src/test/kotlin/sample/kotlin/xyz/srclab/spring/boot/lang/LangSample.kt
package sample.kotlin.xyz.srclab.spring.boot.lang

import org.slf4j.LoggerFactory
import org.springframework.beans.factory.annotation.Value
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.test.context.testng.AbstractTestNGSpringContextTests
import org.testng.Assert
import org.testng.annotations.Test
import xyz.srclab.common.codec.aes.toAesKey
import xyz.srclab.spring.boot.lang.EncodeString

@SpringBootTest(classes = [Starter::class])
open class LangSample : AbstractTestNGSpringContextTests() {

    @Value("AES,BASE64:rliqBhMdiKQDcH8lqNZdIg==")
    private lateinit var encodeString: EncodeString

    @Test
    fun testEncodeString() {
        log.info("encodeString: {}", encodeString)
        val key = "123".toAesKey()
        Assert.assertEquals(encodeString.decodeString(key), "some password")
    }

    companion object {
        private val log = LoggerFactory.getLogger(LangSample::class.java)
    }
}

@SpringBootApplication
open class Starter